NBitcoin/TransactionBuilder.cs

```diff
@@ -1962,8 +2018,8 @@ private Script CreateScriptSig(TransactionSigningContext ctx, ICoin coin, Indexe
 			{
 				if(extension.CanGenerateScriptSig(scriptPubKey))
 				{
-					var scriptSig1 = extension.GenerateScriptSig(scriptPubKey, keyRepo, signer);
-					var scriptSig2 = extension.GenerateScriptSig(scriptPubKey, signer2, signer2);
+					var scriptSig1 = extension.GenerateScriptSig(scriptPubKey, keyRepo, signer, preimageRepo, txIn.TxIn.Sequence);
+					var scriptSig2 = extension.GenerateScriptSig(scriptPubKey, signer2, signer2, preimageRepo, txIn.TxIn.Sequence);

```

Comment:
I don't think that `txIn.TxIn.Sequence` can ever be useful to an extension.

Reply:
It is not necessary, but I think it is useful in the case of Miniscript extension
because txbuilder fails to sign when
1. script contains OP_CSV. and
2. user forgets to set Sequence appropriately.

without this, users will notice that they have signed to the invalid tx when they try to broadcast.

Reply:
(To make me clear, the user does not have to set nSequence even if OP_CSV is in the script if he tries to redeem through non-timelocked path)

Reply:
But don't you have this problem in all case anyway? The transaction might still not be broadcastable depending on the Sequence, and the only way to know is to have the previous UTXOs's height.

Reply:
Yes, TX might be un-broadcastable anyway, difference is that if the user signs to nSequence which is not suitable to pass `OP_CSV` (e.g. disable flag is set) that signature will be useless for redeeming through timelocked redeem condition. so the tx wont be valid forever.